src/components/create-trip/hotel-details.tsx

```diff
@@ -207,8 +208,71 @@ const HotelDetails: React.FC<Props> = ({ stepfn, control }) => {
             <path d="M5 12l6 -6" />
           </svg>
         </Button>
-        <Button type="submit" className="w-[70%]">
-          Submit
+        <Button type="submit" disabled={btnState} className="w-[70%]">
+          {btnState ? (
+            <svg
```

Comment:
Use the loader icon from the icon library and add the 'animate-spin' class to it.